pkg/querier/multi_tenant_querier.go

```diff
@@ -84,17 +86,18 @@ func (q *MultiTenantQuerier) SelectSamples(ctx context.Context, params logql.Sel
 	}
 	params.Selector = updatedSelector.String()
 
-	iters := make([]iter.SampleIterator, len(matchedTenants))
-	i := 0
+	p := pool.NewWithResults[iter.SampleIterator]().WithContext(ctx).WithCancelOnError()
 	for id := range matchedTenants {
-		singleContext := user.InjectOrgID(ctx, id)
-		iter, err := q.Querier.SelectSamples(singleContext, params)
-		if err != nil {
-			return nil, err
-		}
-
-		iters[i] = NewTenantSampleIterator(iter, id)
-		i++
+		id := id
+		p.Go(func(ctx context.Context) (iter.SampleIterator, error) {
```

Comment:
`iter.MapErr` would be a little nicer but it does not work with a context and cancel on error. There's one (https://github.com/sourcegraph/conc/pull/114/) in the work.